MariaCraciun1_Task2/DynamicStringArray.cpp

```diff
@@ -0,0 +1,72 @@
+    #include "DynamicStringArray.hpp"
+
+    DynamicStringArray::DynamicStringArray() : size {0}, dynamicArray {nullptr} {}
+    
+    DynamicStringArray::DynamicStringArray(const DynamicStringArray& obj) {
+        size = obj.getLength() ;
+        dynamicArray = new std::string[size] ;
+        for(int i = 0; i < size; i++ ) {
+            dynamicArray[i] = obj.dynamicArray[i] ;
+        }
+    }
+    
+    DynamicStringArray &DynamicStringArray::operator = (const DynamicStringArray &obj) {
+        if ( this != &obj ){
+            size = obj.getLength() ;
+            dynamicArray = new std::string[size] ;
+             for(int i = 0; i < size; i++ ){
+                dynamicArray[i] = obj.dynamicArray[i] ;
+            }
+        }
+     return *this ;       
+    }
+
+    int DynamicStringArray::getLength () const { 
+        return size;
+    }
+   
+    void DynamicStringArray::addEntry (const std::string &str ) {
+        std::string *tempArray = new std::string[size+1] ;
+        for(int i = 0; i < size; i++ ) {
+            tempArray[i] = dynamicArray[i] ; 
+        }
+        tempArray[size] = str ;
+        size++ ;
+        dynamicArray = tempArray ;
+    }
+    
+    bool DynamicStringArray::deleteEntry (const std::string &str) {
+
+        bool status = false;
+        if ( dynamicArray[0] == str && size == 1 ) { 
+            delete[] dynamicArray ;
+            dynamicArray = nullptr ;
+            size-- ;
+            status = true;
+        }
+        std::string *tempArray = new std::string[size-1] ;
+        int j=0 ;
+        for ( int i=0; i<size; i++){
+            if ( dynamicArray[i] != str )
+                tempArray[j] = dynamicArray[i] ;
+                j++ ;
+        }
+        dynamicArray = tempArray ;
+        size-- ;
+
+        return status ;
+    }
+
+    std::string *DynamicStringArray::getEntry (const int &index) {
```

Comment:
passing const int by reference doesn't really make sense. Usually, non-trivial types to be passed by reference where copy construction is expensive

Reply:
Point taken. But I don't see the harm in leaving it like this. Please let me know if there is something I am missing.

Reply:
well, under the hood it does harm a little bit, some extra instructions(assuming the compiler is not smart enough to ignore it). more info: https://www.generacodice.com/en/articolo/1258791/C%2B%2B%3A-Why-pass-by-value-is-generally-more-efficient-than-pass-by-reference-for-built-in-(i.e.%2C-C-like)-types